__tests__/app.test.js

```diff
@@ -246,3 +246,66 @@ describe("POST", () => {
     })
   });
 });
+
+//========================PATCH=======================
+
+describe('PATCH /api/articles/article_id', () => {
+  test('STATUS 200 should respond with the updated article object', () => {
+    const votesToUpdate = {
+      inc_votes : 10
+    }
+    return request(app)
+    .patch("/api/articles/5")
+    .send(votesToUpdate)
+    .expect(200)
+    .then((response) => {
+      const { updatedArticle } = response.body
+      expect(updatedArticle[0]['article_id']).toBe(5)
+      expect(updatedArticle[0]['votes']).toBe(10)
+      expect(updatedArticle).toEqual([
+        {
+          article_id: 5,
+          title: 'UNCOVERED: catspiracy to bring down democracy',
+          topic: 'cats',
+          author: 'rogersop',
+          body: 'Bastet walks amongst us, and the cats are taking arms!',
+          created_at: '2020-08-03T13:14:00.000Z',
+          votes: 10,
+          article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+        }
+      ])
+    })
+
+  });
+  test('STATUS 404 should respond with an appropriate message when no content is sent with a patch request', () => {
+    return request(app)
+    .patch("/api/articles/3")
+    .send()
+    .expect(404)
+    .then((response)=> {
+    expect(response.body.msg).toBe("Not Found")
+    })
+  });
+  test('STATUS 404 should respond with an appropriate message when no content is sent with a patch request', () => {
+    const votesToUpdate = {}
+    return request(app)
+    .patch("/api/articles/3")
+    .send(votesToUpdate)
+    .expect(404)
+    .then((response)=> {
+      expect(response.body.msg).toBe("Not Found")
+    })
+  })
+  test('STATUS 400 should respond with an appopriate message when given an invalid endpoint', () => {
+    const votesToUpdate = {
+      inc_votes: 20
+    }
+    return request(app)
+    .patch("/api/articles/not-an-id")
+    .send(votesToUpdate)
+    .expect(400)
+    .then((response)=> {
+      expect(response.body.msg).toBe("Bad Request")
+    })
```

Comment:
I'd love to see a 404 for a valid article id that doesn't exist. And update your malformed body tests to the appropriate error code.

Reply:
I agree. I also missed a test for valid endpoint but article_id does not exist. I have added it now. Thank you very much for the comments.